src/main/java/net/imagej/ops/deconvolve/RichardsonLucyNonCirculantC.java

```diff
@@ -0,0 +1,382 @@
+		final long[] end = new long[k.numDimensions()];
+
+		for (int d = 0; d < k.numDimensions(); d++) {
+			start[d] = 0;
+			end[d] = start[d] + k.dimension(d) - 1;
+		}
+
+		copy.compute1(Views.interval(getRAIExtendedEstimate(), new FinalInterval(
+			start, end)), out);
+
+	}
+
+	/**
+	 * create the normalization image needed for semi noncirculant model see
+	 * http://bigwww.epfl.ch/deconvolution/challenge2013/index.html?p=doc_math_rl
+	 */
+	protected void createNormalizationImageSemiNonCirculant(
+		final RandomAccessibleInterval<K> kernel)
+	{
+
+		// k is the window size (valid image region)
+		final int length = k.numDimensions();
+
+		final long[] n = new long[length];
+		final long[] nFFT = new long[length];
+
+		// n is the valid image size plus the extended region
+		// also referred to as object space size
+		for (int d = 0; d < length; d++) {
+			n[d] = k.dimension(d) + l.dimension(d) - 1;
+		}
+
+		for (int d = 0; d < length; d++) {
+			nFFT[d] = getRAIExtendedReblurred().dimension(d);
+		}
+
+		FinalDimensions fd = new FinalDimensions(nFFT);
+
+		// create the normalization image
+		normalization = create.compute1(fd);
+
+		// size of the measurement window
+		final Point size = new Point(length);
+		final long[] sizel = new long[length];
+
+		for (int d = 0; d < length; d++) {
+			size.setPosition(k.dimension(d), d);
+			sizel[d] = k.dimension(d);
+		}
+
+		// starting point of the measurement window when it is centered in fft space
+		final Point start = new Point(length);
+		final long[] startl = new long[length];
+		final long[] endl = new long[length];
+
+		for (int d = 0; d < length; d++) {
+			start.setPosition((nFFT[d] - k.dimension(d)) / 2, d);
+			startl[d] = (nFFT[d] - k.dimension(d)) / 2;
+			endl[d] = startl[d] + sizel[d] - 1;
+		}
+
+		// size of the object space
+		final Point maskSize = new Point(length);
+		final long[] maskSizel = new long[length];
+
+		for (int d = 0; d < length; d++) {
+			maskSize.setPosition(Math.min(n[d], nFFT[d]), d);
+			maskSizel[d] = Math.min(n[d], nFFT[d]);
+		}
+
+		// starting point of the object space within the fft space
+		final Point maskStart = new Point(length);
+		final long[] maskStartl = new long[length];
+
+		for (int d = 0; d < length; d++) {
+			maskStart.setPosition((Math.max(0, nFFT[d] - n[d]) / 2), d);
+			maskStartl[d] = (Math.max(0, nFFT[d] - n[d]) / 2);
+		}
+
+		final RandomAccessibleInterval<O> temp = Views.interval(normalization,
+			new FinalInterval(startl, endl));
+		final Cursor<O> normCursor = Views.iterable(temp).cursor();
+
+		// draw a cube the size of the measurement space
+		while (normCursor.hasNext()) {
+			normCursor.fwd();
+			normCursor.get().setReal(1.0);
+		}
+
+		final Img<O> tempImg = create.compute1(fd);
+
+		// 3. correlate psf with the output of step 2.
+		// TODO: Discuss how to initialize this and whether FFT filters should be
+		// unary...
+		ops().run(CorrelateFFTC.class, tempImg, normalization, kernel,
```

Comment:
we certainly should initialize this somehow. calling `ops().run` is really expensive.